workflow/controller/exec_control.go

```diff
@@ -75,8 +75,8 @@ func (woc *wfOperationCtx) applyExecutionControl(ctx context.Context, pod *apiv1
 	}
 
 	for _, c := range woc.findTemplate(pod).GetMainContainerNames() {
-		if woc.wf.Spec.Shutdown != "" {
-			if _, onExitPod := pod.Labels[common.LabelKeyOnExit]; !woc.wf.Spec.Shutdown.ShouldExecute(onExitPod) {
+		if woc.execWf.Spec.Shutdown != "" {
```

Comment:
can you please add `woc.GetShutdownStrategy()` and `woc.IsSuspend()` to hide this logic?